src/indexing/EventIndexPeg.ts

```diff
@@ -29,6 +29,12 @@ import { SettingLevel } from "../settings/SettingLevel";
 
 const INDEX_VERSION = 1;
 
+/**
+ * Holds the global instance of the `EventIndex` to use across the codebase
+ * (singleton). Looking for an `EventIndex`? Just look for the `EventIndexPeg`
+ * on the peg board. "Peg" is the literal meaning of something you hang
+ * something on. So you'll find a `EventIndex` hanging on the `EventIndexPeg`.
+ */
 export class EventIndexPeg {
     public index: EventIndex = null;
     public error: Error = null;
```

Comment:
not related to your PR, but we really should fix these. They need to either be readonly or we need to drop the `get()` function. Ideally, it becomes private and consumers use the functions they are provided.

Public getters/setters like this are dangerous because it'll almost certainly cause a problem when someone sets it to `null` mid-session.